src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager3.cs

```diff
@@ -0,0 +1,402 @@
+using Standard;
+using Microsoft.Win32;
+using System.Collections;
+using System.Collections.Generic;
+using System.Windows.Interop;
+using System.Windows.Media;
+using System.Windows.Appearance;
+
+namespace System.Windows;
+
+public static class ThemeManager3
+{
+    internal static void OnSystemThemeChanged()
+    {
+        if(IsFluentThemeEnabled)
+        {
+            bool useLightColors = GetUseLightColors(Application.Current.Theme);
+            var fluentThemeResourceUri = GetFluentThemeResourceUri(useLightColors);
+            AddOrUpdateThemeResources(Application.Current.Resources, fluentThemeResourceUri);
+            
+            foreach(Window window in Application.Current.Windows)
+            {
+                if(window == null) continue;
+
+                if(window.Theme == "None")
+                {
+                    ApplyStyleOnWindow(window, useLightColors);
+                    continue;
+                }
+
+                bool windowUseLightColors = GetUseLightColors(window.Theme);
+                var windowFluentThemeResourceUri = GetFluentThemeResourceUri(windowUseLightColors);
+                AddOrUpdateThemeResources(window.Resources, windowFluentThemeResourceUri);
+                ApplyStyleOnWindow(window, windowUseLightColors);
+            }
+        }
+        else
+        {
+            foreach(Window window in _fluentEnabledWindows)
+            {
+                if(window == null || window.IsDisposed) continue;
```

Comment:
at some point windows should be removed from _fluentEnabledWindows if they are no longer valid